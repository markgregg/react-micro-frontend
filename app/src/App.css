
:root {
  --universalMaxListHeight: 600px;
  --color: white;
  --backgroundColor: black;
  --fontSize: small;
}

.frame {
  height: 100vh;
  width: 95vw;
}

.page {
  margin-left: 10%;
  margin-right: 10%;
  width: 80%;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.body {
  display: inline-block;
  position: absolute;
  max-width: 1200px;
  width: 100%;
  font-family: "Cabin";
  font-size: medium;
  background-color: var(--pageColor1);
  color: var(--pageFont);
}

.header {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
}

.title {
  width: 100%;
  text-align: center;
  margin-block-start: 0.3em;
  margin-block-end: 0.1em;
}

.statement {
  font-weight: 200;
}

.menu-bar {
  display: flex;
  flex-direction: row;
  width: 100%;
  padding-top: 15px;
  padding-bottom: 15px;
  border-radius: 8px;
  background-color: var(--pageColor2);
}

.menu {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  gap: 10px;
  flex: 1;
}

.menu-item {
  width: 80px;
  font-size: medium;
  font-weight: bold;
  text-align: center;
  color: var(--universalSelectFontColor);
}

.menu-item:hover {
  color: var(--compactSelectFonHighlightColor,darkgray);
}

.menu-text {
  margin-block-start: 0em;
  margin-block-end: 0em;
}

.context {
  font-weight: 400;
  width: 100%;
  height: 100%;
}

.theme {
  display: block;
  align-self: flex-end;
  --compactSelectBorder: none;
  --compactSelectBackgroundColor: var(--pageColor2);
  --compactSelectHighlightedBackgroundColor: var(--pageColor3);
}

.clipboardCopy {
  font-weight: bold;
}

.footer {
  display: flex;
  flex-direction: column;
  align-items: center;
  border-radius: 8px;
  background-color: var(--pageColor2);
  color: var(--universalSelectFontColor);
}

.footerText {
  margin-block-start: 1em;
  margin-block-end: 1em;
}