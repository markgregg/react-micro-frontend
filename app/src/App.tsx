import CompactSelect from "compact-select";
import React, { useState, useEffect } from "react";
import "./App.css";
import { applyTheme, themes, Themes } from "./themes/themes";
import Examples from "micro-front-end-component1";
import Demo from "micro-front-end-component2";
import About from "micro-front-end-component3";
import { bondList } from "./data/bondList";
import { bigChoices, words, typed, complex} from "./data/data";
import "ag-grid-community/styles/ag-grid.css";
import "ag-grid-community/styles/ag-theme-alpine.css";

const pages = ["About", "Examples", "Demo"];
 
const App = () => {
  const [themeName, setThemeName] = useState<string>(Themes.None.toString());
  const [page, setPage] = useState<string>("Examples");
  
  useEffect(()=> {
    applyTheme(Themes.None);
  },[])

  const setTheme = (theme: string[]) => {
    setThemeName(theme[0]);
    applyTheme(theme[0]);
  }

  return (
    <div className="frame">
      <div className="page">
        <div className="body">
          <div className="header">
            <h1 className="title">An example micro frontend app for React</h1>
            <p className="statement">
              A pratcial working example of how to create a micro frontend react app.
            </p>
          </div>
          <div className="menu-bar">
            <div className="menu">
              {pages.map((pg) => (
                <div
                  key={pg}
                  className="menu-item"
                  onClick={() => setPage(pg)}
                >
                  <p className="menu-text">{pg}</p>
                </div>
              ))}
            </div>
            <div className="theme">
              <CompactSelect
                maximumSelections={1}
                minimumSelections={1}
                selectType="dropdown"
                title="themes"
                choices={themes}
                selected={themeName}
                onChange={setTheme}
                width="80px"
              />
            </div>
          </div>
          <div className="context">
            {(page === "Examples" && <Examples bigChoices={bigChoices} words={words} typed={typed} complex={complex}/>) ||
              (page === "About" && <About/>) ||
              (page === "Demo" && <Demo bondList={bondList}/>)}
          </div>
          <div className="footer">
            <p className="footerText">Created by Mark Gregg</p>
          </div>
        </div>
      </div>
    </div>
  );
};

export default App;
