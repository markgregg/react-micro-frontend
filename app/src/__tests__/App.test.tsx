import ShallowRenderer from 'react-test-renderer/shallow';
import App from '../App';

test("Examples snapshot", () => {
  const renderer = ShallowRenderer.createRenderer();
  renderer.render(
    <App/>
  );
  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});