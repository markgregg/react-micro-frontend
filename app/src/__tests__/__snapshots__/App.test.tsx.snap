// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Examples snapshot 1`] = `
<div
  className="frame"
>
  <div
    className="page"
  >
    <div
      className="body"
    >
      <div
        className="header"
      >
        <h1
          className="title"
        >
          Ag Grid Universal Filter
        </h1>
        <p
          className="statement"
        >
          A powerful and flexible select control that provide a simple mechanism to filter Ag-Grid
        </p>
      </div>
      <div
        className="menu-bar"
      >
        <div
          className="menu"
        >
          <div
            className="menu-item"
            onClick={[Function]}
          >
            <p
              className="menu-text"
            >
              About
            </p>
          </div>
          <div
            className="menu-item"
            onClick={[Function]}
          >
            <p
              className="menu-text"
            >
              Examples
            </p>
          </div>
          <div
            className="menu-item"
            onClick={[Function]}
          >
            <p
              className="menu-text"
            >
              Demo
            </p>
          </div>
        </div>
        <div
          className="theme"
        >
          <CompactSelect
            choices={
              [
                "None",
                "Dark",
                "Light",
                "Blue",
              ]
            }
            maximumSelections={1}
            minimumSelections={1}
            onChange={[Function]}
            selectType="dropdown"
            selected="None"
            title="themes"
            width="80px"
          />
        </div>
      </div>
      <div
        className="context"
      >
        <Examples
          bigChoices={
            [
              "SMITH",
              "JOHNSON",
              "WILLIAMS",
              "BROWN",
              "JONES",
              "MILLER",
              "DAVIS",
              "GARCIA",
              "RODRIGUEZ",
              "WILSON",
              "MARTINEZ",
              "ANDERSON",
              "TAYLOR",
              "THOMAS",
              "HERNANDEZ",
              "MOORE",
              "MARTIN",
              "JACKSON",
              "THOMPSON",
              "WHITE",
              "LOPEZ",
              "LEE",
              "GONZALEZ",
              "HARRIS",
              "CLARK",
              "LEWIS",
              "ROBINSON",
              "WALKER",
              "PEREZ",
              "HALL",
              "YOUNG",
              "ALLEN",
              "SANCHEZ",
              "WRIGHT",
              "KING",
              "SCOTT",
              "GREEN",
              "BAKER",
              "ADAMS",
              "NELSON",
              "HILL",
              "RAMIREZ",
              "CAMPBELL",
              "MITCHELL",
              "ROBERTS",
              "CARTER",
              "PHILLIPS",
              "EVANS",
              "TURNER",
              "TORRES",
              "PARKER",
              "COLLINS",
              "EDWARDS",
              "STEWART",
              "FLORES",
              "MORRIS",
              "NGUYEN",
              "MURPHY",
              "RIVERA",
              "COOK",
              "ROGERS",
              "MORGAN",
              "PETERSON",
              "COOPER",
              "REED",
              "BAILEY",
              "BELL",
              "GOMEZ",
              "KELLY",
              "HOWARD",
              "WARD",
              "COX",
              "DIAZ",
              "RICHARDSON",
              "WOOD",
              "WATSON",
              "BROOKS",
              "BENNETT",
              "GRAY",
              "JAMES",
              "REYES",
              "CRUZ",
              "HUGHES",
              "PRICE",
              "MYERS",
              "LONG",
              "FOSTER",
              "SANDERS",
              "ROSS",
              "MORALES",
              "POWELL",
              "SULLIVAN",
              "RUSSELL",
              "ORTIZ",
              "JENKINS",
              "GUTIERREZ",
              "PERRY",
              "BUTLER",
              "BARNES",
              "FISHER",
              "HENDERSON",
              "COLEMAN",
              "SIMMONS",
              "PATTERSON",
              "JORDAN",
              "REYNOLDS",
              "HAMILTON",
              "GRAHAM",
              "KIM",
              "GONZALES",
              "ALEXANDER",
              "RAMOS",
              "WALLACE",
              "GRIFFIN",
              "WEST",
              "COLE",
              "HAYES",
              "CHAVEZ",
              "GIBSON",
              "BRYANT",
              "ELLIS",
              "STEVENS",
              "MURRAY",
              "FORD",
              "MARSHALL",
              "OWENS",
              "MCDONALD",
              "HARRISON",
              "RUIZ",
              "KENNEDY",
              "WELLS",
              "ALVAREZ",
              "WOODS",
              "MENDOZA",
              "CASTILLO",
              "OLSON",
              "WEBB",
              "WASHINGTON",
              "TUCKER",
              "FREEMAN",
              "BURNS",
              "HENRY",
              "VASQUEZ",
              "SNYDER",
              "SIMPSON",
              "CRAWFORD",
              "JIMENEZ",
              "PORTER",
              "MASON",
              "SHAW",
              "GORDON",
              "WAGNER",
              "HUNTER",
              "ROMERO",
              "HICKS",
              "DIXON",
              "HUNT",
              "PALMER",
              "ROBERTSON",
              "BLACK",
              "HOLMES",
              "STONE",
              "MEYER",
              "BOYD",
              "MILLS",
              "WARREN",
              "FOX",
              "ROSE",
              "RICE",
              "MORENO",
              "SCHMIDT",
              "PATEL",
              "FERGUSON",
              "NICHOLS",
              "HERRERA",
              "MEDINA",
              "RYAN",
              "FERNANDEZ",
              "WEAVER",
              "DANIELS",
              "STEPHENS",
              "GARDNER",
              "PAYNE",
              "KELLEY",
              "DUNN",
              "PIERCE",
              "ARNOLD",
              "TRAN",
              "SPENCER",
              "PETERS",
              "HAWKINS",
              "GRANT",
              "HANSEN",
              "CASTRO",
              "HOFFMAN",
              "HART",
              "ELLIOTT",
              "CUNNINGHAM",
              "KNIGHT",
              "BRADLEY",
              "CARROLL",
              "HUDSON",
              "DUNCAN",
              "ARMSTRONG",
              "BERRY",
              "ANDREWS",
              "JOHNSTON",
              "RAY",
              "LANE",
              "RILEY",
              "CARPENTER",
              "PERKINS",
              "AGUILAR",
              "SILVA",
              "RICHARDS",
              "WILLIS",
              "MATTHEWS",
              "CHAPMAN",
              "LAWRENCE",
              "GARZA",
              "VARGAS",
              "WATKINS",
              "WHEELER",
              "LARSON",
              "CARLSON",
              "HARPER",
              "GEORGE",
              "GREENE",
              "BURKE",
              "GUZMAN",
              "MORRISON",
              "MUNOZ",
              "JACOBS",
              "OBRIEN",
              "LAWSON",
              "FRANKLIN",
              "LYNCH",
              "BISHOP",
              "CARR",
              "SALAZAR",
              "AUSTIN",
              "MENDEZ",
              "GILBERT",
              "JENSEN",
              "WILLIAMSON",
              "MONTGOMERY",
              "HARVEY",
              "OLIVER",
              "HOWELL",
              "DEAN",
              "HANSON",
              "WEBER",
              "GARRETT",
              "SIMS",
              "BURTON",
              "FULLER",
              "SOTO",
              "MCCOY",
              "WELCH",
              "CHEN",
              "SCHULTZ",
              "WALTERS",
              "REID",
              "FIELDS",
              "WALSH",
              "LITTLE",
              "FOWLER",
              "BOWMAN",
              "DAVIDSON",
              "MAY",
              "DAY",
              "SCHNEIDER",
              "NEWMAN",
              "BREWER",
              "LUCAS",
              "HOLLAND",
              "WONG",
              "BANKS",
              "SANTOS",
              "CURTIS",
              "PEARSON",
              "DELGADO",
              "VALDEZ",
              "PENA",
              "RIOS",
              "DOUGLAS",
              "SANDOVAL",
              "BARRETT",
              "HOPKINS",
              "KELLER",
              "GUERRERO",
              "STANLEY",
              "BATES",
              "ALVARADO",
              "BECK",
              "ORTEGA",
              "WADE",
              "ESTRADA",
              "CONTRERAS",
              "BARNETT",
              "CALDWELL",
              "SANTIAGO",
              "LAMBERT",
              "POWERS",
              "CHAMBERS",
              "NUNEZ",
              "CRAIG",
              "LEONARD",
              "LOWE",
              "RHODES",
              "BYRD",
              "GREGORY",
              "SHELTON",
              "FRAZIER",
              "BECKER",
              "MALDONADO",
              "FLEMING",
              "VEGA",
              "SUTTON",
              "COHEN",
              "JENNINGS",
              "PARKS",
              "MCDANIEL",
              "WATTS",
              "BARKER",
              "NORRIS",
              "VAUGHN",
              "VAZQUEZ",
              "HOLT",
              "SCHWARTZ",
              "STEELE",
              "BENSON",
              "NEAL",
              "DOMINGUEZ",
              "HORTON",
              "TERRY",
              "WOLFE",
              "HALE",
              "LYONS",
              "GRAVES",
              "HAYNES",
              "MILES",
              "PARK",
              "WARNER",
              "PADILLA",
              "BUSH",
              "THORNTON",
              "MCCARTHY",
              "MANN",
              "ZIMMERMAN",
              "ERICKSON",
              "FLETCHER",
              "MCKINNEY",
              "PAGE",
              "DAWSON",
              "JOSEPH",
              "MARQUEZ",
              "REEVES",
              "KLEIN",
              "ESPINOZA",
              "BALDWIN",
              "MORAN",
              "LOVE",
              "ROBBINS",
              "HIGGINS",
              "BALL",
              "CORTEZ",
              "LE",
              "GRIFFITH",
              "BOWEN",
              "SHARP",
              "CUMMINGS",
              "RAMSEY",
              "HARDY",
              "SWANSON",
              "BARBER",
              "ACOSTA",
              "LUNA",
              "CHANDLER",
              "BLAIR",
              "DANIEL",
              "CROSS",
              "SIMON",
              "DENNIS",
              "OCONNOR",
              "QUINN",
              "GROSS",
              "NAVARRO",
              "MOSS",
              "FITZGERALD",
              "DOYLE",
              "MCLAUGHLIN",
              "ROJAS",
              "RODGERS",
              "STEVENSON",
              "SINGH",
              "YANG",
              "FIGUEROA",
              "HARMON",
              "NEWTON",
              "PAUL",
              "MANNING",
              "GARNER",
              "MCGEE",
              "REESE",
              "FRANCIS",
              "BURGESS",
              "ADKINS",
              "GOODMAN",
              "CURRY",
              "BRADY",
              "CHRISTENSEN",
              "POTTER",
              "WALTON",
              "GOODWIN",
              "MULLINS",
              "MOLINA",
              "WEBSTER",
              "FISCHER",
              "CAMPOS",
              "AVILA",
              "SHERMAN",
              "TODD",
              "CHANG",
              "BLAKE",
              "MALONE",
              "WOLF",
              "HODGES",
              "JUAREZ",
              "GILL",
              "FARMER",
              "HINES",
              "GALLAGHER",
              "DURAN",
              "HUBBARD",
              "CANNON",
              "MIRANDA",
              "WANG",
              "SAUNDERS",
              "TATE",
              "MACK",
              "HAMMOND",
              "CARRILLO",
              "TOWNSEND",
              "WISE",
              "INGRAM",
              "BARTON",
              "MEJIA",
              "AYALA",
              "SCHROEDER",
              "HAMPTON",
              "ROWE",
              "PARSONS",
              "FRANK",
              "WATERS",
              "STRICKLAND",
              "OSBORNE",
              "MAXWELL",
              "CHAN",
              "DELEON",
              "NORMAN",
              "HARRINGTON",
              "CASEY",
              "PATTON",
              "LOGAN",
              "BOWERS",
              "MUELLER",
              "GLOVER",
              "FLOYD",
              "HARTMAN",
              "BUCHANAN",
              "COBB",
              "FRENCH",
              "KRAMER",
              "MCCORMICK",
              "CLARKE",
              "TYLER",
              "GIBBS",
              "MOODY",
              "CONNER",
              "SPARKS",
              "MCGUIRE",
              "LEON",
              "BAUER",
              "NORTON",
              "POPE",
              "FLYNN",
              "HOGAN",
              "ROBLES",
              "SALINAS",
              "YATES",
              "LINDSEY",
              "LLOYD",
              "MARSH",
              "MCBRIDE",
              "OWEN",
              "SOLIS",
              "PHAM",
              "LANG",
              "PRATT",
              "LARA",
              "BROCK",
              "BALLARD",
              "TRUJILLO",
              "SHAFFER",
              "DRAKE",
              "ROMAN",
              "AGUIRRE",
              "MORTON",
              "STOKES",
              "LAMB",
              "PACHECO",
              "PATRICK",
              "COCHRAN",
              "SHEPHERD",
              "CAIN",
              "BURNETT",
              "HESS",
              "LI",
              "CERVANTES",
              "OLSEN",
              "BRIGGS",
              "OCHOA",
              "CABRERA",
              "VELASQUEZ",
              "MONTOYA",
              "ROTH",
              "MEYERS",
              "CARDENAS",
              "FUENTES",
              "WEISS",
              "HOOVER",
              "WILKINS",
              "NICHOLSON",
              "UNDERWOOD",
              "SHORT",
              "CARSON",
              "MORROW",
              "COLON",
              "HOLLOWAY",
              "SUMMERS",
              "BRYAN",
              "PETERSEN",
              "MCKENZIE",
              "SERRANO",
              "WILCOX",
              "CAREY",
              "CLAYTON",
              "POOLE",
              "CALDERON",
              "GALLEGOS",
              "GREER",
              "RIVAS",
              "GUERRA",
              "DECKER",
              "COLLIER",
              "WALL",
              "WHITAKER",
              "BASS",
              "FLOWERS",
              "DAVENPORT",
              "CONLEY",
              "HOUSTON",
              "HUFF",
              "COPELAND",
              "HOOD",
              "MONROE",
              "MASSEY",
              "ROBERSON",
              "COMBS",
              "FRANCO",
              "LARSEN",
              "PITTMAN",
              "RANDALL",
              "SKINNER",
            ]
          }
          complex={
            [
              {
                "description": "in some villages, you can't find one tree",
                "name": "tedious",
                "value": 101,
              },
              {
                "description": "was shocked that in villages",
                "name": "tank",
                "value": 102,
              },
              {
                "description": "Africa desertification is one of the reasons",
                "name": "stingy",
                "value": 103,
              },
              {
                "description": "more likely to have been cut down for construction",
                "name": "treat",
                "value": 104,
              },
              {
                "description": "he began to raise money to make his",
                "name": "wish",
                "value": 105,
              },
              {
                "description": "engage with communities across the region",
                "name": "wrathful",
                "value": 106,
              },
              {
                "description": "knew would be able to rise to the challenge",
                "name": "crown",
                "value": 107,
              },
              {
                "description": "village and there's no women",
                "name": "push",
                "value": 108,
              },
              {
                "description": "passion to plant trees by helping women",
                "name": "manage",
                "value": 109,
              },
              {
                "description": "seeds and what to do to grow the",
                "name": "ritzy",
                "value": 110,
              },
              {
                "description": "market and we are not dependent",
                "name": "stream",
                "value": 111,
              },
              {
                "description": "have an account in the bank now",
                "name": "dark",
                "value": 112,
              },
              {
                "description": "from palms and tamarinds",
                "name": "deer",
                "value": 113,
              },
              {
                "description": "planting still needs to be done",
                "name": "bad",
                "value": 114,
              },
              {
                "description": "been key in helping organise funds",
                "name": "treatment",
                "value": 115,
              },
              {
                "description": "they work in a tiny workshop with",
                "name": "things",
                "value": 116,
              },
              {
                "description": "machines to make them from local",
                "name": "shiny",
                "value": 117,
              },
              {
                "description": "it's the symbol of an African tree",
                "name": "female",
                "value": 118,
              },
              {
                "description": "island in the middle of the river",
                "name": "quiver",
                "value": 119,
              },
              {
                "description": "and in some cases even underneath",
                "name": "cow",
                "value": 120,
              },
              {
                "description": "meant the water's edge was much",
                "name": "bomb",
                "value": 121,
              },
              {
                "description": "if it gets worse it could mean leaving",
                "name": "muddled",
                "value": 122,
              },
              {
                "description": "salty or fresh water, have now been",
                "name": "unusual",
                "value": 123,
              },
              {
                "description": "planted along the shoreline",
                "name": "fish",
                "value": 124,
              },
              {
                "description": "promise, one day of fruits and even shade",
                "name": "sisters",
                "value": 125,
              },
              {
                "description": "we are not dependent",
                "name": "bat",
                "value": 126,
              },
              {
                "description": "raise money to make his dream",
                "name": "aftermath",
                "value": 127,
              },
              {
                "description": "organising the mass planting",
                "name": "vengeful",
                "value": 128,
              },
              {
                "description": "desertification is one of the reasons",
                "name": "actor",
                "value": 129,
              },
              {
                "description": "from his own pocket to kick-start",
                "name": "nippy",
                "value": 130,
              },
            ]
          }
          typed={
            [
              {
                "text": "tedious",
                "value": "101",
              },
              {
                "text": "tank",
                "value": "102",
              },
              {
                "text": "stingy",
                "value": "103",
              },
              {
                "text": "treat",
                "value": "104",
              },
              {
                "text": "wish",
                "value": "105",
              },
              {
                "text": "wrathful",
                "value": "106",
              },
              {
                "text": "crown",
                "value": "107",
              },
              {
                "text": "push",
                "value": "108",
              },
              {
                "text": "manage",
                "value": "109",
              },
              {
                "text": "ritzy",
                "value": "110",
              },
              {
                "text": "stream",
                "value": "111",
              },
              {
                "text": "dark",
                "value": "112",
              },
              {
                "text": "deer",
                "value": "113",
              },
              {
                "text": "bad",
                "value": "114",
              },
              {
                "text": "treatment",
                "value": "115",
              },
              {
                "text": "things",
                "value": "116",
              },
              {
                "text": "shiny",
                "value": "117",
              },
              {
                "text": "female",
                "value": "118",
              },
              {
                "text": "quiver",
                "value": "119",
              },
              {
                "text": "cow",
                "value": "120",
              },
              {
                "text": "bomb",
                "value": "121",
              },
              {
                "text": "muddled",
                "value": "122",
              },
              {
                "text": "unusual",
                "value": "123",
              },
              {
                "text": "fish",
                "value": "124",
              },
              {
                "text": "sisters",
                "value": "125",
              },
              {
                "text": "bat",
                "value": "126",
              },
              {
                "text": "aftermath",
                "value": "127",
              },
              {
                "text": "vengeful",
                "value": "128",
              },
              {
                "text": "actor",
                "value": "129",
              },
              {
                "text": "nippy",
                "value": "130",
              },
            ]
          }
          words={
            [
              "tedious",
              "tank",
              "stingy",
              "treat",
              "wish",
              "wrathful",
              "crown",
              "push",
              "manage",
              "ritzy",
              "stream",
              "dark",
              "deer",
              "bad",
              "treatment",
              "things",
              "shiny",
              "female",
              "quiver",
              "cow",
              "bomb",
              "muddled",
              "unusual",
              "fish",
              "sisters",
              "bat",
              "aftermath",
              "vengeful",
              "actor",
              "nippy",
            ]
          }
        />
      </div>
      <div
        className="footer"
      >
        <p>
          Created by Mark Gregg
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`VerticalMenu snapshot 1`] = `
<div
  className="frame"
>
  <div
    className="page"
  >
    <div
      className="body"
    >
      <div
        className="header"
      >
        <h1
          className="title"
        >
          Ag Grid Universal Filter
        </h1>
        <p
          className="statement"
        >
          A powerful and flexible select control that provide a simple mechanism to filter Ag-Grid
        </p>
      </div>
      <div
        className="menu-bar"
      >
        <div
          className="menu"
        >
          <div
            className="menu-item"
            onClick={[Function]}
          >
            <p
              className="menu-text"
            >
              About
            </p>
          </div>
          <div
            className="menu-item"
            onClick={[Function]}
          >
            <p
              className="menu-text"
            >
              Examples
            </p>
          </div>
          <div
            className="menu-item"
            onClick={[Function]}
          >
            <p
              className="menu-text"
            >
              Demo
            </p>
          </div>
        </div>
        <div
          className="theme"
        >
          <div
            className="csCompactSelect compactSelectStandard"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onPaste={[Function]}
            style={
              {
                "height": undefined,
                "maxHeight": undefined,
                "maxWidth": undefined,
                "minHeight": undefined,
                "minWidth": undefined,
                "width": "80px",
              }
            }
          >
            <div
              className="toolTip"
            >
              <div
                className="csContentSelectWrapper"
              >
                <div
                  className="csCompactMainDisplay"
                >
                  <div
                    className="csCompactTextWrapper"
                  >
                    <p
                      className="csCompactSelectTextDisplay csCompactSelectTextDisplaySelectedItem"
                      style={{}}
                    />
                  </div>
                  <div>
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      strokeWidth="0"
                      style={
                        {
                          "color": undefined,
                        }
                      }
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g>
                        <path
                          d="M0 0h24v24H0z"
                          fill="none"
                        />
                        <path
                          d="M12 14l-4-4h8z"
                        />
                      </g>
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="context"
      >
        <div
          className="examples"
        >
          <div
            className="vmenu"
          >
            <h3
              className="vmenu-heading"
            >
              Catagories
            </h3>
            <ul
              className="vmenu-items"
            >
              <li
                className="option"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  {
                    "backgroundColor": "var(--pageColor2)",
                  }
                }
              >
                <p>
                  Binding
                </p>
              </li>
              <li
                className="option"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  {
                    "backgroundColor": "var(--pageColor2)",
                  }
                }
              >
                <p>
                  Single Select
                </p>
              </li>
              <li
                className="option"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  {
                    "backgroundColor": "var(--pageColor2)",
                  }
                }
              >
                <p>
                  Multi Select
                </p>
              </li>
              <li
                className="option"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  {
                    "backgroundColor": "var(--pageColor2)",
                  }
                }
              >
                <p>
                  Look-up
                </p>
              </li>
              <li
                className="option"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  {
                    "backgroundColor": "var(--pageColor2)",
                  }
                }
              >
                <p>
                  Custom Values
                </p>
              </li>
              <li
                className="option"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  {
                    "backgroundColor": "var(--pageColor2)",
                  }
                }
              >
                <p>
                  Selection prefixed
                </p>
              </li>
              <li
                className="option"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  {
                    "backgroundColor": "var(--pageColor2)",
                  }
                }
              >
                <p>
                  Disabled
                </p>
              </li>
              <li
                className="option"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  {
                    "backgroundColor": "var(--pageColor2)",
                  }
                }
              >
                <p>
                  Ag-Grid
                </p>
              </li>
            </ul>
          </div>
          <div
            className="controls"
          />
        </div>
      </div>
      <div
        className="footer"
      >
        <p>
          Created by Mark Gregg
        </p>
      </div>
    </div>
  </div>
</div>
`;
